src/transformStateWithClones.js

```diff
@@ -7,7 +7,26 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
+
+      for (const key of action.keysToRemove) {
+        delete currentState[key];
+      }
+    }
```

Comment:
[CODE STYLE]: Use a switch statement instead of multiple if/else if conditions when you have a limited amount of conditions to make the code cleaner and more maintainable. Also, include a default case for error handling. Example:

switch (action.type) {
  case 'clear':
    // ...
    break;
  case 'addProperties':
    // ...
    break;
  case 'removeProperties':
    // ...
    break;
  default:
    // handle unknown action type
}

Reply:
Fix it 
<img width="956" alt="image" src="https://github.com/user-attachments/assets/699ec71d-139a-4cbb-8b4f-979ef3dda7e3">